### PR TITLE
Test CMFGEN submodule using Pytest Arraydiff

### PR DESCRIPTION
**Pull request by atharva-2001**
### :pencil: Description

**Type:** :vertical_traffic_light: `testing` 

This PR restructures CMFGEN tests and also adds tests for the [utils.py](https://github.com/tardis-sn/carsus/blob/master/carsus/io/cmfgen/util.py) file from the CMFGEN submodule using Pytest Arraydiff.

Changes made:
- Test functions in [utils.py](https://github.com/tardis-sn/carsus/blob/master/carsus/io/cmfgen/util.py) using pytest arraydiff.
- Remove text blocks used to compare dataframes and instead return dataframes using pytest arraydiff. 
- Remove fixtures which only serve the purpose of providing the path of the reference file. There were quite a few such fixtures, and I felt we were repeating ourselves. I am parameterising a path fixture with different file names now. 
- Add a step to the test workflow file to run pytest arraydiff tests.

For running tests locally:
- For just running tests: `pytest carsus --refdata=refdata/path -x --arraydiff --arraydiff-reference-path=refdata/arraydiff/path`
- For generating reference files: `pytest carsus --refdata=refdata/path -x --arraydiff-generate-path=refdata/arraydiff/path`

### :pushpin: Resources

Examples, notebooks, and links to useful references.

### :vertical_traffic_light: Testing

How did you test these changes?

- [X] Testing pipeline
- [ ] Other method (describe)
- [ ] My changes can't be tested (explain why)

### :ballot_box_with_check: Checklist

- [x] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes
- [ ] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

**Comment by epassaro** [comment]
branch is called `main` not `master` @atharva-2001

File: carsus/io/cmfgen/util.py [hunk 1]
```diff
@@ -270,7 +270,7 @@ def get_opproject_phixs_table(threshold_energy_ryd, a, b, c, d, e, n_points=1000
     return phixs_table
 
 
-def get_hummer_phixs_table(threshold_energy_ryd, a, b, c, d, e, f, g, h, n_points=1000):
+def get_hummer_phixs_table(threshold_energy_ryd, a, b, c, d, e, f, g, n_points=1000):
     """
```
Comment (atharva-2001) [inline carsus/io/cmfgen/util.py:274]: The `h` parameter was never used.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -90,8 +98,13 @@ jobs:
       - name: Install package
         run: pip install -e .
 
+      - name: Generate CMFGEN reference data
```
[the remark this answers is not in the document]
Comment (epassaro) [reply .github/workflows/tests.yml:101]: Exactly.

[the remark this answers is not in the document]
Comment (atharva-2001) [reply .github/workflows/tests.yml]: Done!

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -92,6 +100,7 @@ jobs:
 
       - name: Run tests
         run: pytest carsus ${{ env.PYTEST_FLAGS }}
+             --arraydiff --arraydiff-reference-path=carsus-refdata/arraydiff
```
Comment (epassaro) [inline .github/workflows/tests.yml:103]: Add those lines to the flags variable

**Review by epassaro — APPROVED** [review]

File: carsus/io/cmfgen/util.py [hunk 4]
```diff
@@ -270,7 +270,7 @@ def get_opproject_phixs_table(threshold_energy_ryd, a, b, c, d, e, n_points=1000
     return phixs_table
 
 
-def get_hummer_phixs_table(threshold_energy_ryd, a, b, c, d, e, f, g, h, n_points=1000):
```
Comment (epassaro) [inline carsus/io/cmfgen/util.py:273]: We were aware that the model in the paper has 8 parameters but for some reason, the function here is using just 7. I don't know if we should remove the unused variable or not.

File: .github/workflows/tests.yml (hunk 3 above)
Comment (atharva-2001) [reply .github/workflows/tests.yml]: Done in https://github.com/tardis-sn/carsus/pull/311/commits/2dca529e05d5563f75d0403e060a4348acffe7ed

**Pull request merged**